packages/variable-resolver/src/browser/variable-resolver-service.ts

```diff
@@ -109,13 +109,13 @@ export class VariableResolverService {
     }
 
     protected async resolveVariables(value: string, context: VariableResolverService.Context): Promise<void> {
+        const variableRegExp = new RegExp(VariableResolverService.VAR_REGEXP);
```

Comment:
Could you elaborate why do we need to wrap one instance of `RegExp` into another

`/pattern/attributes` is an equivalent of `new RegExp(pattern, attributes)`

Reply:
I have changed it because  i have noticed that if I run few tasks **in parallel** then the "options.cwd" of some tasks were not resolved correctly ( and were still ${workspacefolder}).

And after my change, when i run few tasks in parallel , the resolves works well.

The reason for that is that the RegExp object is stateful  and that you can not share a regexp
  object to parse several strings in parallel.

please see  : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec

![image](https://user-images.githubusercontent.com/18531202/70236904-914ff800-176e-11ea-97da-7f72e66e8f54.png)